tenant/repository.go

```diff
@@ -39,6 +42,15 @@ func (s DBService) Exists(tenantID uuid.UUID) bool {
 	return true
 }
 
+func (s DBService) ExistsWithNsUsername(nsUsername string) bool {
+	var t Tenant
+	err := s.db.Table(t.TableName()).Where("ns_username = ?", nsUsername).Find(&t).Error
+	if err != nil {
+		return false
+	}
+	return true
```

Comment:
Check if the error is NotFoundError. If yes then return false. Otherwise return the error. Otherwise if there is some problem with the query or DB different than expected Not Found then you will 1) get an infinite loop when constructing an unique name 2) miss opportunity to catch/log some problem with DB.

Reply:
Good point - I blindly copied the approach from the [Exists](https://github.com/fabric8-services/fabric8-tenant/blob/master/tenant/repository.go#L33) method that was already there and I didn't realize these cases.
I'll send another PR to fix the `Exists` method as well.